### ISSUE TITLE
Vestibule Circulation / "Pick-up Only"

### ISSUE THREAD

**Issue by ejk** [aadl/arborcat#116]
Hold shelves by the door. Zoom station for staff interaction.

Hold shelf is staff only. Need to notify how they want to pick it up:

- Locker pickup
- Vestibule pickup (current day + one extra day)
- I Don't want this

Email with links for various actions. Add contact info for if you have trouble without explicitly mentioning mail service.

Send a second email when item is checked out and put in the vestibule or in the locker.

Change the hold status for the different stages.

Lockers should be a form to select from pre-set time slots (3 slots). Collect a phone number for a text or a call when it's ready. 1 locker per patron per time slot.

No tools items at first.

**Comment by ejk** [comment]
Table for Patron Pickup selections

- hold_id (primary key)
- patron_id (int)
- branch (char OR varchar OR enum)
- pickup_date (date)
- time_slot (int, use 0 for vestibule pickup, time slot # for locker pickup)
- phone (int, optional)
- completed (tinyint, default 0, set to 1 when hold is checked out and placed for pickup)

**Comment by lauriek334** [comment]
Here's the updated workflow for hold pickups based on yesterday's conversations and feedback. [pickup workflow.pdf](https://github.com/aadl/arborcat/files/4703137/pickup.workflow.pdf)

**Comment by ulotrichous** [comment]
Same-day won't be an option in any case. So for locker pickups, we'll only offer available times starting the next day from the moment the form is opened; and for vestibule pickups, the earliest choice would be tomorrow.

A design question that can be informed by the resulting dev complexity is, do we offer them any day for the next week as their vestibule pickup date? Or just tomorrow? Or just tomorrow or the day after? etc.

**Comment by eby** [comment]
The site should have access to the holding shelf expire date since it would be the normal hold expires. I think it makes sense to offer the days up to that minus one or whatever. 

Otherwise if I know I want to pick it up on Wednesday and I'm only offered Monday it is going to end up as staff interaction and work. I'm not going to want to try to remember to do it on Tuesday.

**Comment by ulotrichous** [comment]
OK, how about this, the horizon for both locker bookings and vestibule pickup is the day *after* the hold would have expired. So if they act on the last day that they have to act on their hold, then can still choose tomorrow. Or we could make it 7 days from current, which would give people more flexbility at the cost of some items sitting 2 weeks before being picked up.

**Comment by ulotrichous** [comment]
Another feature we will need; support for more than one vestibule per location. At westgate we think we'll have the front vestibule and the back vestibule and they can choose which.

**Comment by taleon** [comment]
Yeah, I was planning to have a config set up per location where we can easily add/modify pickup locations like that. The list printing will process each one per location on its own pages, and the locker ones will be separated by the time slots.

I think allowing that extra day works. I know people are eager to be picking up materials, so I don't know if we'd want to be holding popular items for another week, but that's just my thought.
